haystack/utils/deepsetcloud.py

```diff
@@ -717,6 +723,11 @@ def _transition_pipeline_state(
                 f"Pipeline config '{pipeline_config_name}' is in invalid state '{status.value}' to be transitioned to '{target_state.value}'."
             )
 
+        if status in failed_states:
+            logger.warning(
+                f"Last status of '{pipeline_config_name}' was: '{status}'. Retrying to reach the state '{target_state}'. "
```

Comment:
I think the term "retrying" is misleading here since the method will just run and wont actively run the same logic multiple times. It is rather a second (manual) try. I updated the waring to: 

`"Pipeline config '{pipeline_config_name}' is in a failed state '{status}'. This might be caused by a previous error during (un)deployment. Trying to transition from '{status}' to '{target_state}'..."`

What do you think ?